### PR TITLE
docs(site): Group nav items into 'Guides' and 'Components'

### PR DESCRIPTION
**Pull request by markdalgleish**
## Commit Message for Review

This change is paving the way for more documentation by providing some basic structure to the IA. As part of this change, I've standardised the page titles and URLs for component demo pages to strictly match the name of the component (e.g. `Text` instead of `Typography`, especially since we're likely to write a high level guide to typography at some point)

### CODE REVIEW

**Comment by markdalgleish** [comment]
The new IA looks like this:

screen shot 2017-08-15 at 1 27 24 pm

File: docs/src/components/App/Header/Header.js [hunk 1]
```diff
@@ -61,17 +61,35 @@ export default class Header extends Component {
...
-                          </Text>
-                        ))
-                      }
+                      <Card transparent>
+                        <Text headline regular><Link className={styles.link} to="/">Home</Link></Text>
+                      </Card>
+
+                      <Card transparent>
+                        <h2>
+                          <Text h2 headline>Guides</Text>
```
Comment (michaeltaranto) [inline docs/src/components/App/Header/Header.js]: You seem to have a rogue `h2` prop here.

Comment (markdalgleish) [inline docs/src/components/App/Header/Header.js]: lol, looks like you spotted my little API experiment...

Comment (markdalgleish) [inline docs/src/components/App/Header/Header.js]: Just removed this 👍

**Pull request merged**